docs/README.rst

```diff
@@ -0,0 +1,106 @@
+F5 ML2 Driver for OpenStack
+===========================
+
+.. sidebar:: **OpenStack version:**
+
+   |openstack|
+
+|Build Status|
+
+.. raw:: html
+
+   <script async defer src="https://f5-openstack-slack.herokuapp.com/slackin.js"></script>
+
+.. toctree:
+   titlesonly
+   hidden
+
+
+version |release|
+-----------------
+
+|release-notes|
+
+
+Architecture
+------------
+
+The |ml2-long| is an OpenStack Neutron `ML2 Mechanism Driver`_. It provides the means for the |agent-long| to correctly bind Neutron ports to BIG-IP resources.
+
+When you create a new Neutron loadbalancer, the |ml2| does the following:
+
+- checks whether the allocated Neutron port belongs to an F5 device, and
+- looks for an active F5 Agent.
+
+If it finds an active F5 Agent, the |ml2| checks the agent configuration to find out if the advertised tunnel type(s) matches the type of the requested port. If it doesn't find a matching tunnel type, the |ml2| checks the physical network bridge mappings (:code:`f5_external_physical_mappings`) for a match. The |ml2| will not bind the port if it doesn't find a match in either of these Agent settings. Likewise, if it doesn't find an Agent configuration file, the |ml2| won't bind the port.
+
+When the |ml2| does find a match for the requested Neutron port in the the |agent| configuration file, it adds the following metadata to the port:
+
+- :code:`device-owner = "network:f5lbaasv2"`
+- :code:`host_id = <f5-agent_uuid>`
```

Comment:
I would say the the host_id corresponds to the agent ID, this is the agent_id configuration item in f5-openstack-agent.ini file.